src/routing_table/mod.rs

```diff
@@ -806,8 +806,16 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
                     // this is to prevent spamming the network by sending messages with
                     // intentionally short prefixes
                     if prefix.is_covered_by(self.prefixes().iter()) {
-                        return Ok(self.iter()
-                            .filter(|name| prefix.matches(name) && **name != self.our_name)
+                        let is_compatible = |(pfx, section)| if prefix.is_compatible(pfx) {
+                            Some(section)
+                        } else {
+                            None
+                        };
+                        return Ok(self.sections
+                            .iter()
+                            .filter_map(is_compatible)
+                            .flat_map(BTreeSet::iter)
+                            .chain(self.our_section.iter().filter(|name| **name != self.our_name))
```

Comment:
do we need to change the comment of If the destination is an Authority::PrefixSection part, as now is not limited to our_prefix?